### PR TITLE
[CMR] Various Menu Fixes

### PR DESCRIPTION
**Pull request by WilkinsKa1**
## Description

This PR addresses the following:
• the aria-active-descendant focus that was happening that Jay asked to remove
• the viewport display issues John had mentioned earlier this week (if interested, [more details](https://github.com/reach/reach-ui/commit/dae686dfd3943f70087195115b97ca50e81e97f9#diff-15ead492ca6c18618236cb9abce33372) on that one)
• partially addresses the "Menu" text on the mobile nav not opening the menu. We get consistent interactions with that button now (Yay!) but part of fixing this was realizing the isOpen bool was not really doing what we thought it was doing. More on this below.

## What still needs to be done
- [x] You'll notice we no longer see the overlay or icon change when the menu is toggled open. As mentioned, this wasn't actually doing anything, which was leading to inconsistent behavior. I've begun looking into solutions so we can have these again.
- [x] Need to be able to close nested menus when they are clicked

## Type of Change
- Bug fix ('fix')

### CODE REVIEW

**Review by Jskobos — COMMENTED** [review]
Menu behavior is much better this way. Please take a look at the related stuff in #6871 and let me know what you think...my understanding after reading the docs is that the MenuLink in the MobileNav should be a MenuItem, and we should be passing a callback to the onSelect rather than the onClick. I'm not sure how exactly that'll work out in this particular component, but it helped a lot with the normal action menus.

**Comment by Jskobos** [comment]
I have no idea what the patch was doing for menu-button so I'm not qualified on whether we need to remove or update the patch to get the build to pass.

**Comment by Jskobos** [comment]
OK, changed my mind about this; using MenuLink in here is better.

File: packages/manager/src/components/PrimaryNav/MobileNav.tsx [hunk 1]
```diff
@@ -150,116 +149,121 @@ const useStyles = makeStyles((theme: Theme) => ({
...
 export const MobileNav: React.FC<Props> = props => {
   const classes = useStyles();
+
   const ref = React.useRef<HTMLDivElement>(null);
   const { groups } = props;
 
-  const [isOpen, setIsOpen] = React.useState<boolean>(false);
-
-  React.useMemo(() => {
-    if (isOpen) {
```
Comment (WilkinsKa1) [inline packages/manager/src/components/PrimaryNav/MobileNav.tsx:165]: Note to self: explore using `isExpanded` to manage our own state so we can control display of the nested menus

File: packages/manager/src/components/PrimaryNav/MobileNav.tsx [hunk 2]
```diff
@@ -157,113 +157,98 @@ const useStyles = makeStyles((theme: Theme) => ({
...
-                <div className={classes.menuWrapper}>
-                  {groups.map((thisGroup: any) => {
-                    // For each group, filter out hidden links.
-                    const filteredLinks = thisGroup.links.filter(
-                      (thisLink: any) => !thisLink.hide
-                    );
-                    if (filteredLinks.length === 0) {
-                      return null;
-                    }
+    <ReachMenu key={window.location.pathname}>
```
Comment (johnwcallahan) [inline packages/manager/src/components/PrimaryNav/MobileNav.tsx:163]: Most of the diff is from a linter fix. The big change is the addition of this `key`, and one more below.

File: packages/manager/src/components/PrimaryNav/MobileNav.tsx [hunk 3]
```diff
@@ -157,113 +157,98 @@ const useStyles = makeStyles((theme: Theme) => ({
...
-    </>
+                          {filteredLinks.map((thisLink: any) => (
+                            <MenuLink
+                              data-testid={`menu-item-${thisLink.display}`}
+                              key={thisLink.display}
+                              as={Link}
+                              to={thisLink.href}
+                              className={classes.menuItemLink}
+                              disabled={
+                                window.location.pathname === thisLink.href
```
Comment (johnwcallahan) [inline packages/manager/src/components/PrimaryNav/MobileNav.tsx:234]: This is the other big change.

**Comment by johnwcallahan** [comment]
Just pushed some changes to this.

TLDR: our patch doesn't work anymore, so we're stuck with double menu issue, originally addressed in https://github.com/linode/manager/pull/6523. I tried applying the patch again, but was not successful. I asked for an update on the Reach UI Issue (linked in https://github.com/linode/manager/pull/6523), so we'll see what they say. I think we'll likely need to wait for them to fix it.

I also fixed an issue where clicking a nested menu item (like "Compute" -> "Linodes") didn't close the menu. To do this I used `window.location.pathname` as a `key`, which forces the component to re-render when the route changes.

The caveat here is that this doesn't work when you're _already on_ the clicked route, so to address this, I disabled the currently active route. Curious to hear your feedback on this one.

**Review by tiffwong — APPROVED** [review]
Woohoo! Can confirm the menu button in the mobile nav works properly now 😅

**Review by Jskobos — COMMENTED** [review]
There are a few things here I'm not sure about, most of which are probably preexisting conditions. Going to spell them out before I approve though:
(all wrt MobileNav only)
1. If I'm on a submenu, I can hit ESC to close it, but pressing ESC doesn't then close the menu (ESC does close the menu if you never opened a submenu in the first place)
2. I can't seem to tab or key my way through the submenus. Pressing the down arrow moves me from Dashboard to Marketplace and that's about all I can do.

**Review by Jskobos — APPROVED** [review]

**Comment by johnwcallahan** [comment]
> There are a few things here I'm not sure about, most of which are probably preexisting conditions. Going to spell them out before I approve though:
> (all wrt MobileNav only)
> 
> 1. If I'm on a submenu, I can hit ESC to close it, but pressing ESC doesn't then close the menu (ESC does close the menu if you never opened a submenu in the first place)
> 2. I can't seem to tab or key my way through the submenus. Pressing the down arrow moves me from Dashboard to Marketplace and that's about all I can do.

We chatted offline, we're going to address these points separately.
